janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java

```diff
@@ -224,6 +224,7 @@ private synchronized void closeInternal() {
              */
             for (StandardJanusGraphTx otx : openTransactions) {
                 try {
+                    otx.rollback();
```

Comment:
What about thread-bounded transactions?
Documentation says:

> If transactions are still open when close() is called, then the behavior of the outstanding transactions is technically undefined. In practice, any non-thread-bound transactions will usually be effectively rolled back, but the thread-bound transaction belonging to the thread that invoked shutdown will first be committed. Note, that both read and write operations occur within the context of a transaction.

Reply:
>  I think auto-close mode is not clear here. I don't think that commit all stale txs when graph is closed is good idea (in case CLOSE_BEHAVIOR.COMMIT configured)

Why do you think it isn't a good idea to commit transactions when `CLOSE_BEHAVIOR.COMMIT` is configured?

Reply:
Auto-close mode is good for try-resource pattern, but when we close graph some tx may partially done

Example, tx configured as auto-clode commit
```
try (tx) {
  write
  read
  // <-- here graph is closing
  write 
} // <-- client expect auto-commit here   
```
And if we follow configured auto-close mode, some data may be inconsistence. But if we rollback, we return to consistent state

Anyway, is bad practice, close graph during any open transaction exists

Reply:
Thank you for the explanation. I agree with you that it is a bad practice but if we always call `rollback` on closing graph then:
1. Documentation must be changed to describe new behavior (`docs/badics.adoc` section `Transaction Handling`). As I understand `CLOSE_BEHAVIOR.COMMIT` will not work anymore.
2. If it is a breaking change that it should be targeted `master` branch. And the breaking change must be described in `docs/upgrade.adoc`

Reply:
> Documentation must be changed to describe new behavior (docs/badics.adoc section Transaction Handling). As I understand CLOSE_BEHAVIOR.COMMIT will not work anymore.

No. In any case CLOSE_BEHAVIOR work as expected, except during graph is closed. Moreover, master implemented behavior not conform documentation of [JanusGraph.newTransaction](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java#L111)

>      * The transaction is open when it is returned but MUST be explicitly closed by calling {@link org.janusgraph.core.JanusGraphTransaction#commit()}
>      * or {@link org.janusgraph.core.JanusGraphTransaction#rollback()} when it is no longer needed. 

Would be good if any one else make review these changes

Reply:
Thank you for the explanation @mad . In such case the changes looks good to me